src/main/java/de/triology/cas/services/RegistryEtcd.java

```diff
@@ -77,6 +79,59 @@ public String getFqdn() {
         }
     }
 
+    public URI getCasLogoutUri(String doguname) throws GetCasLogoutUriException {
+        JSONObject doguMetaData;
+        try {
+            EtcdKeysResponse.EtcdNode node = getDoguNodeFromEtcd(doguname);
+            doguMetaData = getCurrentDoguNode(node);
+            JSONObject properties;
+            if (doguMetaData != null) {
+                properties = getPropertiesFromMetaData(doguMetaData);
+            } else {
+                throw new GetCasLogoutUriException("Could not get dogu metadata");
+            }
+            return getLogoutUriFromProperties(properties);
+        } catch (JsonException | NullPointerException | ParseException | URISyntaxException | GetDoguNodeFromEtcdException e) {
+            throw new GetCasLogoutUriException(e.toString());
+        }
+    }
+
+    private URI getLogoutUriFromProperties(JSONObject properties) throws GetCasLogoutUriException, URISyntaxException {
+        Object logoutUri = properties.get("logoutUri");
+        if (logoutUri != null){
+            String logoutUriString = logoutUri.toString();
+            if (logoutUriString != null) {
+                return new URI(logoutUriString);
+            } else {
+                throw new GetCasLogoutUriException("Could not get logoutUri from properties");
+            }
+        } else {
+            throw new GetCasLogoutUriException("Could not get logoutUri from properties");
+        }
+    }
+
+    private JSONObject getPropertiesFromMetaData(JSONObject doguMetaData) throws JsonException, NullPointerException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'groovy.json.JsonException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.NullPointerException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)